tools/@aws-cdk/remodel/lib/template/packages/aws-cdk-lib/scripts/gen.ts

```diff
@@ -1,7 +1,50 @@
 import { generateAll } from '@aws-cdk/cfn2ts';
+import { ModuleDefinition } from '@aws-cdk/pkglint';
+import * as fs from 'fs-extra';
 import * as path from 'path';
 
-const srcDir = path.join(__dirname, '..', 'lib');
-generateAll(srcDir, {
-  coreImport: '../../core',
-});
+const awsCdkLibDir = path.join(__dirname, '..');
+const srcDir = path.join(awsCdkLibDir, 'lib');
+
+main()
+  .then(() => process.exit(0))
+  // eslint-ignore-next-line no-console
+  .catch(console.error)
+
+async function main() {
+  const generated = await generateAll(srcDir, {
+    coreImport: '../../core',
+  });
+
+  addExportsToPackageJson(generated);
+  addModulesToIndexFile(generated);
+}
+
+function addExportsToPackageJson(modules: ModuleDefinition[]) {
+  const pkgJsonPath = path.join(awsCdkLibDir, 'package.json');
+  const pkgJson = fs.readJsonSync(pkgJsonPath);
+  modules.forEach((module) => {
+    // Add export to the package.json if it's not there yet.
+    if (!pkgJson.exports[`./${module.moduleName}`]) {
+      pkgJson.exports[`./${module.moduleName}`] = `./${module.moduleName}/index.js`;
```

Comment:
in package.json we need to include `lib` since its relative to there.

```suggestion
      pkgJson.exports[`./${module.moduleName}`] = `./lib/${module.moduleName}/index.js`;
```